### PR TITLE
HDDS-2653. Improve executor memory usage in new Freon tests

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

Let all Freon threads run test tasks in a loop instead of submitting a job for each task.  This avoids memory issues caused by too many jobs.

https://issues.apache.org/jira/browse/HDDS-2653

## How was this patch tested?

Ran a few types of Freon tests (`omkg`, `ockg`, `ockv`) with both low and high `-n` values (low for testing normal completion, high for validating fix for OOME).

### CODE REVIEW

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/BaseFreonGenerator.java [hunk 1]
```diff
@@ -101,71 +101,102 @@
...
+  }
 
-      //provider is usually a lambda, print out only the owner class name:
-      String spanName = provider.getClass().getSimpleName().split("\\$")[0];
+  /**
+   * Launches {@code threadNo} task runners in executor.  Each one executes test
+   * tasks in a loop until completion or failure.
+   */
+  private void startTaskRunners(TaskProvider provider) {
+    for (int i = 0; i < threadNo; i++) {
```
Comment (xiaoyuyao) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/BaseFreonGenerator.java]: NIT: threadNo is confusing here as we are submit task with no guaranteed # of threads?

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/BaseFreonGenerator.java [hunk 2]
```diff
@@ -212,6 +249,13 @@ public void init() {
           }
           printReport();
         }));
+
+    executor = Executors.newFixedThreadPool(threadNo);
+
+    progressBar = new ProgressBar(System.out, testNo, successCounter::get);
```
Comment (xiaoyuyao) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/BaseFreonGenerator.java]: We need to ensure the pgoressBar thread is shutdown at the end.

**Comment by xiaoyuyao** [comment]
LGTM, +1. Thanks @adoroszlai  for the improvement.

**Comment by adoroszlai** [comment]
Thanks @xiaoyuyao for reviewing and committing this.

**Pull request merged**